.github/workflows/ci_workflows.yml

```diff
@@ -87,16 +82,16 @@ jobs:
             toxenv: py38-test-alldeps-numpy118-cov-clocale
             toxposargs: --remote-data=astropy
```

Comment:
Those 2 builds seem redondant since "oldestdeps" includes Numpy 1.18 ?

Reply:
@saimn , are you saying I should move `-cov-clocale` to oldest deps job?

Reply:
Hmm actually one difference is "oldestdeps" vs "alldeps-numpy118". Maybe "oldestdeps-alldeps-cov-clocale" would work ?
Also for a proper coverage I think we would need `-cov` on the Windows and MacOS jobs. No sure how overhead is added by the coverage measure.

Reply:
Re: coverage -- I think @astrofrog was opposed to going overboard with the coverage because the CI takes a while to run as it is.

Reply:
Yeah sure. But then better if we can remove a job :).

Reply:
Oh and it seems the (recently added I think) thread safety tests are quite slow : 
Windows:
```
21.06s call     .tox/py39-test-alldeps/lib/site-packages/astropy/utils/tests/test_decorators.py::test_classproperty_lazy_threadsafe
20.95s call     .tox/py39-test-alldeps/lib/site-packages/astropy/utils/tests/test_decorators.py::test_lazyproperty_threadsafe
20.60s call     .tox/py39-test-alldeps/lib/site-packages/astropy/units/tests/test_quantity_ufuncs.py::TestUfuncHelpers::test_thread_safety
```
MacOS:
```
15.38s call     .tox/py39-test-alldeps/lib/python3.9/site-packages/astropy/utils/tests/test_decorators.py::test_lazyproperty_threadsafe
15.26s call     .tox/py39-test-alldeps/lib/python3.9/site-packages/astropy/utils/tests/test_decorators.py::test_classproperty_lazy_threadsafe
12.27s call     .tox/py39-test-alldeps/lib/python3.9/site-packages/astropy/units/tests/test_quantity_ufuncs.py::TestUfuncHelpers::test_thread_safety
```

Reply:
threadsafety test performance should be a new issue.

They were added by @bmerry in #11224 and #11226.

Reply:
Those tests would be good candidates for  #12350.